src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -604,39 +599,25 @@ private void ParseStatusLine(Span<byte> line, HttpResponseMessage response)
                 (HttpStatusCode)(100 * (status1 - '0') + 10 * (status2 - '0') + (status3 - '0'));
 
             // Parse (optional) reason phrase
-            byte c = line[12];
-            if (c == '\r')
+            if (line.Length == 12)
             {
                 response.ReasonPhrase = string.Empty;
             }
-            else if (c != ' ')
+            else if (line[12] != ' ')
             {
                 ThrowInvalidHttpResponse();
             }
             else
             {
-                Span<byte> reasonBytes = line.Slice(13, line.Length - 13 - 2); // 2 == \r\n ending trimmed off
+                Span<byte> reasonBytes = line.Slice(13);
                 string knownReasonPhrase = HttpStatusDescription.Get(response.StatusCode);
                 if (knownReasonPhrase != null && EqualsOrdinal(knownReasonPhrase, reasonBytes))
                 {
                     response.ReasonPhrase = knownReasonPhrase;
                 }
                 else
                 {
-                    unsafe
-                    {
-                        try
-                        {
-                            fixed (byte* reasonPtr = &MemoryMarshal.GetReference(reasonBytes))
-                            {
-                                response.ReasonPhrase = Encoding.ASCII.GetString(reasonPtr, reasonBytes.Length);
-                            }
-                        }
-                        catch (FormatException e)
-                        {
-                            ThrowInvalidHttpResponse(e);
-                        }
-                    }
+                    response.ReasonPhrase = Encoding.ASCII.GetString(reasonBytes);
```

Comment:
Nice to see the new overloads being used :)

Reply:
This could also now be condensed if desired:
```C#
response.ReasonPhrase = knownReasonPhrase != null && EqualsOrdinal(knownReasonPhrase, reasonBytes) ?
    knownReasonPhrase :
    Encoding.ASCII.GetString(reasonBytes);
```

Also, is ASCII technically correct?  I see elsewhere we use:
https://github.com/dotnet/corefx/blob/c54cf8989aebe539ac9c7929fea14c4a0b62c721/src/System.Net.Http/src/System/Net/Http/HttpRuleParser.cs#L43-L47
but maybe for the status line ASCII is correct, or maybe it's close enough that it doesn't matter?

Reply:
iso-8859-1 seems to be preferred, per RFC.  I wonder if we have other places we are using ASCII that should be iso-8859-1.